### DISCUSSION TITLE
getting single object without Accept: application/vnd.pgrst.object ?

### DISCUSSION THREAD

**Discussion by tvogt** [PostgREST/postgrest#2338] [Q&A] [answered]
Is there another way to force PostgREST to answer with a single object instead of an array besides setting the "Accept: application/vnd.pgrst.object" header?

I'm trying to serve GeoJSON to a Mapbox-GL-JS app and Mapbox overwrites custom Accept: headers with application/json. And an array is not a properly formated GeoJSON response.

**Comment by steve-chavez** [comment, marked as answer]
> Mapbox overwrites custom Accept: headers with application/json

Hm, yeah that looks like trouble. Do you know if it's possible to configure Mapbox to specify an `Accept: application/geo+json` header instead? Then we could solve this with https://github.com/PostgREST/postgrest/pull/1564.

**Reply by steve-chavez** [reply]
`application/geo+json` is made official in https://datatracker.ietf.org/doc/html/rfc7946. I believe the correct behavior for Mapbox client would be to accept the `application/geo+json` type with a higher `q`factor([ref](https://developer.mozilla.org/en-US/docs/Glossary/Quality_values)) than `application/json`.

**Reply by steve-chavez** [reply]
I see you reported https://github.com/mapbox/mapbox-gl-js/issues/12011, I think it would be reasonable to propose to change their `Accept` to have q factors and include `application/geo+json`.

**Reply by tvogt** [reply]
I agree that mapbox should fix their code.

I'd still like another option to serving singular from PostgREST. There are other clients out there, and who knows when mapbox will update their code. In fact, from an API perspective, having a global config option for this sounds to me a good idea.

**Reply by steve-chavez** [reply]
Assuming we have the geojson feature(merging https://github.com/PostgREST/postgrest/pull/1564), then we'd need to provide a transformation from `application/json` to `application/geo+json`. Likely through https://github.com/PostgREST/postgrest/pull/1582#issuecomment-1154431651.

Another quicker option would be to add a proxy(like Nginx) in front of PostgREST, that does the same thing - transforming the header. @tvogt Would that be possible for you?

**Reply by steve-chavez** [reply]
(I can help with merging #1564, but realistically #1582 is going to take much longer.)

**Reply by steve-chavez** [reply]
I think there might be a simpler solution, check my suggestion on https://github.com/PostgREST/postgrest/discussions/2339#discussioncomment-2975453

**Reply by tvogt** [reply]
@steve-chavez I'd prefer not adding a proxy just to transform a header.

I think defining functions instead of views is the best approach, because it also lets me use filters. Something that doesn't seem to work when using a view defined as above, at least I couldn't get any filters to work except "?type=eq.FeatureCollection" :-/

Not sure if it's possible to embed related tables on a function, though. So there's another challenge.
